### PR TITLE
Fixing issue 9402

### LINKED ISSUE #9402: Default form is displayed for a Publication when I specified a proper submission-name
**Describe the bug**
Not all collections are being presented the correct configured form. 
It you have such a configuration like this in you item-submission.xml file:
```xml
<name-map collection-entity-type="Publication" submission-name="Publication" />
```
And have more than 10 collections with `dspace.entity.type = Publication` the 11th will not display the correct form. It will use the default one.

**To Reproduce**
1. You will need more than 10 collections with `dspace.entity.type = Publication` (or any other type)
2. Add more collections and associate the Publication type
3. Deposit one new item on a recent collection

**Expected behavior**
I'm expecting to have the correct form. But the default is shown.

**Related work**
- https://github.com/DSpace/DSpace/pull/9004
- https://github.com/DSpace/DSpace/pull/8855

### PR DESCRIPTION
**Pull request by paulo-graca**
## References
* Fixes #9402

## Description
This, very short PR, refactors `findAllCollectionsByEntityType` to use private method `retrieveCollectionsWithSubmit` with pagination and runs pages to find all collections for a specific entity type.

## Instructions for Reviewers
This issue fixes #9402, the instructions to reproduce the issue are:
* You will need more than 10 collections with dspace.entity.type = Publication (or any other type)
* Add more collections and associate the Publication type
* Deposit one new item on a recent collection

List of changes in this PR:
* Refactors findAllCollectionsByEntityType to use retrieveCollectionsWithSubmit
* adds a new constant `SOLR_ROWS_PER_PAGE` to `CollectionService` that holds the rows number for each page result

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java [hunk 1]
```diff
@@ -1054,23 +1054,45 @@ public int countCollectionsWithSubmit(String q, Context context, Community commu
...
 
-        DiscoverResult discoverResult = searchService.search(context, discoverQuery);
-        List<IndexableObject> solrIndexableObjects = discoverResult.getIndexableObjects();
+            discoverQuery.setStart((page - 1) * SOLR_ROWS_PER_PAGE);
 
-        for (IndexableObject solrCollection : solrIndexableObjects) {
-            Collection c = ((IndexableCollection) solrCollection).getIndexedObject();
-            collectionList.add(c);
+            //retrieve the specific page with results
+            DiscoverResult discoverResult = retrieveCollectionsWithSubmit(context, discoverQuery,
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java:1082]: Overall, this PR appears to work.  But, I'm wondering why we aren't just calling `findCollectionsWithSubmit(null, context, null, entityType, -1, -1)`?   It appears that would immediately return all Collections in one List, without having to use the "for" loop that you have in this code?

If there's a reason why you prefer your approach, that's fine.  I'm just noting that it seems like your logic is the same as the logic would be in `findCollectionsWithSubmit()` if you disabled the offset & limt.

**Comment by kshepherd** [comment]
sorry i haven't finished my review yet, i got distracted while looking into why we need to turn off authorization for the retrieve method? can you explain why this is required @paulo-graca ?
if it is absolutely necessary then we should wrap it in a try-with-resources or try/finally block so we can ensure there is never any chance of the context persisting with elevated privileges beyond the scope of retrieving a collection list.
thanks!

**Comment by tdonohue** [comment]
Alternative approach to this PR is now available in #9478.  @kshepherd : You might be interested in reviewing that alternative approach, so I assigned it to you, @paulo-graca and myself.

**Comment by paulo-graca** [comment]
> sorry i haven't finished my review yet, i got distracted while looking into why we need to turn off authorization for the retrieve method? can you explain why this is required @paulo-graca ? if it is absolutely necessary then we should wrap it in a try-with-resources or try/finally block so we can ensure there is never any chance of the context persisting with elevated privileges beyond the scope of retrieving a collection list. thanks!

@kshepherd I left a comment explaining why. It's related with the solr query that is built during the process. Without it, the query will be built considering only the user that is requesting it. We need the fetch all the results, not only those for that particular user. 

However, as @tdonohue mentioned, a better approach, which validates the configurations at runtime, is implemented in this PR - https://github.com/DSpace/DSpace/pull/9478

**Comment by paulo-graca** [comment]
I will close this PR. https://github.com/DSpace/DSpace/pull/9478 also addresses the issue with a better implementation.

**Pull request closed without merging**